source/Generic/DateImporter/DateImporter.psm1

```diff
@@ -846,5 +883,123 @@ function Get-SteamLicenses
+        $gameDatabase
+    )
+
+    if ($gameDatabase.count -eq 0)
+    {
+        return
+    }
+
+    $libraryName = "Origin"
+    $LicensesList = Get-OriginLicenses $libraryName
+    if ($LicensesList.count -eq 0)
+    {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_LicensesNotFoundMessage"), "$libraryName Date Importer")
+        return
+    }
+
+    Set-DatesFromLicenses $libraryName $LicensesList $gameDatabase
+}
+
+function Get-OriginLicenses
+{
+    param (
+        $libraryName
+    )
+    
+    [System.Collections.Generic.List[object]]$LicensesList = @()
+
+    $authUrl = "https://accounts.ea.com/connect/auth?client_id=ORIGIN_JS_SDK&response_type=token&redirect_uri=nucleus:rest&prompt=none"
+    $loginStatus = Get-LoginStatusViaJson $authUrl
+    if ($loginStatus -eq $false)
+    {
+        $PlayniteApi.Dialogs.ShowMessage(([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_UsetNotLoggedMessage") -f $libraryName), "$libraryName Date Importer")
+        return $LicensesList
+    }
+
+    $webView = $PlayniteApi.WebViews.CreateOffscreenView()
+    $webView.NavigateAndWait($authUrl)
+    $pageSource = $webView.GetPageSource()
+    $webView.Dispose()
+
+    $jsonValid = Get-IsJsonValidFromPage $pageSource
+
+    if($jsonValid -eq $false) 
+    {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_MenuItemImportOriginInvalidJson"), "$libraryName Date Importer")
+        return $LicensesList
+    }
+
+    $authJson = Get-JsonFromPageSource $pageSource
+    $token = $authJson.access_token
+
+
+    $idUrl = "https://gateway.ea.com/proxy/identity/pids/me"
+    $headers = @{
+        Authorization = "Bearer {0}" -f $token
+    }
+    $idResponse = $null
+    try {
+        $idResponse = Invoke-RestMethod -Method "GET" -Uri $idUrl -Headers $headers -ContentType "application/json"
+    } catch {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_MenuItemImportOriginErrorMessage"), "$libraryName Date Importer")
+        return $LicensesList
+    }
+    $id = $idResponse.pid.pidId
+
+
+    $licenseTemplate = "https://api1.origin.com/ecommerce2/consolidatedentitlements/{0}?machine_hash=1"
+    $licenseUrl = $licenseTemplate -f $id 
+    $headers = @{
+        authtoken = $token
+    }
+    $xml = $null
+    try {
+        $xml = Invoke-RestMethod -Method "GET" -Uri $licenseUrl -Headers $headers -ContentType "application/xml"
+    } catch {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_MenuItemImportOriginErrorMessage"), "$libraryName Date Importer")
+        return $LicensesList
+    }
+
+
+    $licenses = Select-Xml $xml -XPath '/entitlements/entitlement'
+    foreach ($license in $licenses) {
+        $type = $license.Node.offer.offerType.ToLower() -replace " ", ""
+        if($($type -eq "basegame") -or $($type -eq "demo")) 
```

Comment:
A space after the `if` here too please

Reply:
Done in 559566c951bf00f312b54bf5f8efed31753055ee.